scripts/sample_dataset_request.py

```diff
@@ -0,0 +1,414 @@
+        """
+        Create the output filename for the dataset.
+
+        Parameters
+        ----------
+        ds
+            Loaded dataset
+
+        Returns
+        -------
+            The output filename
+        """
+        output_path = (
+            Path(os.path.join(*[metadata[item] for item in self.obs4mips_path_items]))
+            / f"v{metadata['version']}"
+        )
+        if ds_filename.name.split("_")[0] == ds.variable_id:
+            filename_prefix = "_".join([metadata[item] for item in self.obs4mips_filename_paths])
+        else:
+            filename_prefix = ds_filename.name.split("_")[0] + "_"
+            filename_prefix += "_".join(
+                [metadata[item] for item in self.obs4mips_filename_paths if item != "variable_id"]
+            )
+
+        if "time" in ds.dims:
+            time_range = (
+                f"{ds.time.min().dt.strftime('%Y%m').item()}-{ds.time.max().dt.strftime('%Y%m').item()}"
+            )
+            filename = f"{filename_prefix}_{time_range}.nc"
+        else:
+            filename = f"{filename_prefix}.nc"
+
+        return output_path / filename
+
+
+def fetch_datasets(DataRequest) -> pd.DataFrame:
+    """
+    Fetch the datasets from ESGF.
+
+    Parameters
+    ----------
+    search_facets
+        Facets to search for
+    remove_ensembles
+        Whether to remove ensembles from the dataset
+        (i.e. include only a single ensemble member)
+
+    Returns
+    -------
+    List of paths to the fetched datasets
+    """
+    cat = ESGFCatalog()
+
+    cat.search(**DataRequest.facets)
+    if DataRequest.remove_ensembles:
+        cat.remove_ensembles()
+
+    path_dict = cat.to_path_dict(prefer_streaming=False, minimal_keys=False)
+    merged_df = cat.df.merge(pd.Series(path_dict, name="files"), left_on="key", right_index=True)
+    if DataRequest.time_span:
+        merged_df["time_start"] = DataRequest.time_span[0]
+        merged_df["time_end"] = DataRequest.time_span[1]
+    print(merged_df)
+    return merged_df
+
+
+def deduplicate_datasets(DataRequest) -> pd.DataFrame:
+    """
+    Deduplicate a dataset collection.
+
+    Uses the metadata from the first dataset in each group,
+    but expands the time range to the min/max timespan of the group.
+
+    Parameters
+    ----------
+    datasets
+        The dataset collection
+
+    Returns
+    -------
+    pd.DataFrame
+        The deduplicated dataset collection spanning the times requested
+    """
+    datasets = fetch_datasets(DataRequest)
+
+    def _deduplicate_group(group: pd.DataFrame) -> pd.DataFrame:
+        first = group.iloc[0].copy()
+        first.time_start = group.time_start.min()
+        first.time_end = group.time_end.max()
+
+        return first
+
+    return datasets.groupby("key").apply(_deduplicate_group, include_groups=False).reset_index()
+
+
+def create_sample_dataset(DataRequest):
```

Comment:
This functions should probably take an argument named `request` with a type hint of `DataRequest`. Otherwise you are clashing with the global `DataRequest` type and we use snake case for variable names.

```suggestion
def create_sample_dataset(request: DataRequest):
```